staging/src/k8s.io/apiserver/pkg/endpoints/handlers/negotiation/negotiate.go

```diff
@@ -140,6 +140,19 @@ func (emptyEndpointRestrictions) AllowsMediaTypeTransform(mimeType string, mimeS
 func (emptyEndpointRestrictions) AllowsServerVersion(string) bool  { return false }
 func (emptyEndpointRestrictions) AllowsStreamSchema(s string) bool { return s == "watch" }
 
+var DiscoveryEndpointRestrictions = discoveryEndpointRestrictions{}
+
+type discoveryEndpointRestrictions struct{}
+
+func (discoveryEndpointRestrictions) AllowsMediaTypeTransform(mimeType string, mimeSubType string, gvk *schema.GroupVersionKind) bool {
+	if gvk != nil {
+		return gvk.Group == "apidiscovery.k8s.io" && gvk.Version == "v2beta1" && gvk.Kind == "APIGroupDiscoveryList"
```

Comment:
Im not familiar with this interface. What is this doing?

Reply:
We do not allow additional Accept info (`Accept: application/json;k=v;...`) to be passed in for a normal endpoint. This interface allows the those values to be passed in and is hard coded as v2beta1 apidiscovery types at the moment.